tests/baselines/reference/nonNullFullInference.types

```diff
@@ -0,0 +1,36 @@
+=== tests/cases/compiler/nonNullFullInference.ts ===
+// https://github.com/microsoft/TypeScript/issues/19577
+
+function testNonNullInference(numbers: number[]) {
+>testNonNullInference : (numbers: number[]) => number
+>numbers : number[]
+
+    let last;
+>last : any
+
+    for (const n of numbers) {
+>n : number
+>numbers : number[]
+
+        if (n % 2) {
+>n % 2 : number
+>n : number
+>2 : 2
+
+            return n;
+>n : number
+        }
+
+        last = n;
+>last = n : number
+>last : any
+>n : number
+    }
+
+    last; // number
+>last : number

```

Comment:
Is this correct though? `numbers` could always be an empty array so `last` could have never been assigned to. The current inferred type for this is `number | undefined` and it reflects the runtime behavior better.

Reply:
It seems either the baseline generation omits `undefined` in the types, or this existed prior to the change being made.